akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/AccountExampleOneLiners.java

```diff
@@ -0,0 +1,157 @@
+
+    interface Account {}
+    public static class EmptyAccount implements Account {}
+    public static class OpenedAccount implements Account {
+        public final double balance;
+
+        OpenedAccount(double balance) {
+            this.balance = balance;
+        }
+    }
+    public static class ClosedAccount implements Account {}
+
+    public static Behavior<AccountCommand> behavior(String accountNumber) {
+        return Behaviors.setup(context -> new AccountExampleOneLiners(context, accountNumber));
+    }
+
+    public AccountExampleOneLiners(ActorContext<AccountCommand> context, String accountNumber) {
+        super(accountNumber);
+    }
+
+    @Override
+    public Account emptyState() {
+        return new EmptyAccount();
+    }
+
+    private Effect<AccountEvent, Account> createAccount() {
+        return Effect().persist(new AccountCreated());
+    }
+
+    private Effect<AccountEvent, Account> depositCommand(Deposit deposit) {
+        return Effect().persist(new Deposited(deposit.amount));
+    }
+
+    private Effect<AccountEvent, Account> withdrawCommand(OpenedAccount account, Withdraw withdraw) {
+        if ((account.balance - withdraw.amount) < 0.0) {
+            return Effect().unhandled(); // TODO replies are missing in this example
+        } else {
+            return Effect().persist(new Withdrawn(withdraw.amount))
+                    .andThen(acc2 -> { // FIXME in scaladsl it's named thenRun, change javadsl also?
+                        // we know this cast is safe, but somewhat ugly
+                        OpenedAccount openAccount = (OpenedAccount) acc2;
+                        // do some side-effect using balance
+                        System.out.println(openAccount.balance);
+                    });
+        }
+    }
+
+    private Effect<AccountEvent, Account> closeCommand(OpenedAccount account, CloseAccount cmd) {
+        if (account.balance == 0.0)
+            return Effect().persist(new AccountClosed());
+        else
+            return Effect().unhandled();
+    }
+
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, EmptyAccount, Account> initialHandler() {
+        return commandHandlerBuilder(EmptyAccount.class)
+                .matchCommand(CreateAccount.class, this::createAccount);
+    }
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, OpenedAccount, Account> openedAccountHandler() {
+        return commandHandlerBuilder(OpenedAccount.class)
+                .matchCommand(Deposit.class, this::depositCommand)
+                .matchCommand(Withdraw.class, this::withdrawCommand)
+                .matchCommand(CloseAccount.class, this::closeCommand);
+    }
+
+    private CommandHandlerBuilder<AccountCommand, AccountEvent, ClosedAccount, Account> closedHandler() {
+        return commandHandlerBuilder(ClosedAccount.class)
+                .matchCommand(AccountCommand.class, () -> Effect().unhandled());
```

Comment:
we should look at what typical compilation errors looks like with many overloaded matchCommand

that can be written with same number of characters like this :-P
```
.matchCommand(AccountCommand.class, __ -> Effect().unhandled());
```

Reply:
👍 true, I learned this recently.